finetuner/experiment.py

```diff
@@ -180,6 +181,82 @@ def create_run(
         )
         return run
 
+    def _prepare_data(
+        self,
+        train_data: Union[DocumentArray, str],
+        eval_data: Union[DocumentArray, str],
+        eval_callback: EvaluationCallback,
+        run_name: str,
+    ):
+        """Upload data to Hubble and returns their names.
+
+        Uploads all data needed for fine-tuning - training data,
+        evaluation data and query/index data for `EvaluationCallback`.
```

Comment:
Is it really possible to use both `eval_data` as well as `query_data` and `index_data` for this callback? I thought the user either provides `eval_data` or the query + index data. I may be wrong.

Reply:
When only `eval_data` is provided, `query_data` becomes `eval_data` and `index_data` is `None` afaik.